tiledb/sm/subarray/range_subset.h

```diff
@@ -141,7 +140,78 @@ struct SortStrategy<std::string, std::string> {
           return a.start_str() < b.start_str() ||
                  (a.start_str() == b.start_str() && a.end_str() < b.end_str());
         });
-    return Status::Ok();
+  };
+};
+
+/** Default merge behavior: merging is not enabled. */
+template <typename T, typename Enable = T>
+struct MergeStrategy {
+  static void merge(std::vector<Range>&);
+};
+
+/** Merge algorithm for numeric-type ranges. */
+template <typename T>
+struct MergeStrategy<
+    T,
+    typename std::enable_if<std::is_arithmetic<T>::value, T>::type> {
+  static void merge(std::vector<Range>& ranges) {
+    auto head = ranges.begin();
+    size_t merged_cells = 0;
+
+    // Merge
+    for (auto tail = head + 1; tail != ranges.end(); tail++) {
+      const bool can_coalesce = *static_cast<const T*>(head->end_fixed()) !=
+                                    std::numeric_limits<T>::max() &&
+                                *static_cast<const T*>(head->end_fixed()) + 1 ==
+                                    *static_cast<const T*>(tail->start_fixed());
+      const bool can_merge = *static_cast<const T*>(head->start_fixed()) <=
+                                 *static_cast<const T*>(tail->start_fixed()) &&
+                             *static_cast<const T*>(tail->start_fixed()) <=
+                                 *static_cast<const T*>(head->end_fixed());
+
+      if (can_coalesce || can_merge) {
+        head->set_end_fixed(tail->end_fixed());
+        merged_cells++;
+      } else {
+        head++;
+        std::swap(*head, *tail);
+      }
+    }
+
+    // Resize
+    ranges.resize(ranges.size() - merged_cells);
+  };
+};
+
+/** Merge algorithm for string-ASCII-type ranges. */
+template <>
+struct MergeStrategy<std::string, std::string> {
+  static void merge(std::vector<Range>& ranges) {
+    auto head = ranges.begin();
+    size_t merged_cells = 0;
+
+    // Merge
+    for (auto tail = head + 1; tail != ranges.end(); tail++) {
```

Comment:
+1 doesn't really apply to string ranges... Remove the coalesce code.

Reply:
Note: as discussed on slack, _char_ ranges should be coalesced, but _string_ ranges should not. Change has been made accordingly.